src/test/scala/com/databricks/spark/avro/AvroSuite.scala

```diff
@@ -382,10 +403,49 @@ class AvroSuite extends FunSuite {
       assert(df.count == 8)
 
       val tempSaveDir = s"$tempDir/save/"
-
-      df.write.avro(tempSaveDir)
+            df.write.avro(tempSaveDir)
       val newDf = TestSQLContext.read.avro(tempSaveDir)
       assert(newDf.count == 8)
     }
   }
-}
\ No newline at end of file
+
+  test("test doc in meta") {
+    val df = TestSQLContext.read.avro(testFile)
+    df.schema.fields(0).metadata.getString(SchemaConverters.METADATA_KEY_DOC)
+
+    for (x <- df.schema.fields) {
+      if (x.name == "title") {
+        assert("episode title" == x.metadata.getString(SchemaConverters.METADATA_KEY_DOC))
+      } else if (x.name == "doctor") {
+        assert("main actor playing the Doctor in episode" ==
+          x.metadata.getString(SchemaConverters.METADATA_KEY_DOC))
+      } else if (x.name == "air_date") {
+        assert("initial date" == x.metadata.getString(SchemaConverters.METADATA_KEY_DOC))
+      }
+    }
+  }
+
+  test("test aliases in meta") {
+    val df = TestSQLContext.read.avro(testFile)
+    df.schema("string").metadata.getStringArray(SchemaConverters.METADATA_KEY_ALIASES) === Array("string_alias1", "string_alias1")
+  }
+
+  test("test aliases columns in data frame") {
+    var df = TestSQLContext.read.avro(testFile)
+    var fieldArray = df.schema.fieldNames;
+    assert(fieldArray contains("string"))
+    assert(!(fieldArray contains("string_alias1")))
+    assert(!(fieldArray contains("string_alias2")))
+    assert(!(fieldArray contains("map_alias")))
+    assert(!(fieldArray contains("enum_alias")))
+    assert(!(fieldArray contains("union_int_alias")))
+
+    fieldArray = SchemaConverters.dataFrameWithAliasColumn(df).schema.fieldNames
```

Comment:
this test should probably use the user visible option instead of diving into internals.